sdk/messaging/azservicebus/example_receiver_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azservicebus_test
+
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus"
+)
+
+func ExampleClient_NewReceiverForQueue() {
+	receiver, err = client.NewReceiverForQueue(
+		queueName,
+		// The receive mode controls when a message is deleted from Service Bus.
+		//
+		// `azservicebus.PeekLock` is the default. The message is locked, preventing multiple
+		// receivers from processing the message at once. You control the lock state of the message
+		//  using one of the message settlement functions, processor.CompleteMessage(), which removes
+		// it from Service Bus, or processor.AbandonMessage(), which makes it available again.
+		//
+		// `azservicebus.ReceiveAndDelete` causes Service Bus to remove the message as soon
+		// as it's received.
+		//
+		// More information about receive modes:
+		// https://docs.microsoft.com/azure/service-bus-messaging/message-transfers-locks-settlement#settling-receive-operations
+		azservicebus.ReceiverWithReceiveMode(azservicebus.PeekLock),
+	)
+
+	exitOnError("Failed to create Receiver", err)
+}
+
+func ExampleClient_NewReceiverForSubscription() {
+	receiver, err = client.NewReceiverForSubscription(
+		topicName,
+		subscriptionName,
+		// The receive mode controls when a message is deleted from Service Bus.
+		//
+		// `azservicebus.PeekLock` is the default. The message is locked, preventing multiple
+		// receivers from processing the message at once. You control the lock state of the message
+		//  using one of the message settlement functions, processor.CompleteMessage(), which removes
+		// it from Service Bus, or processor.AbandonMessage(), which makes it available again.
+		//
+		// `azservicebus.ReceiveAndDelete` causes Service Bus to remove the message as soon
+		// as it's received.
+		//
+		// More information about receive modes:
+		// https://docs.microsoft.com/azure/service-bus-messaging/message-transfers-locks-settlement#settling-receive-operations
```

Comment:
Are these documentation bits included in the docs for each method? I don't think these should be in the examples and should be in the docstring for the methods. If they're in both then godoc will render both and have repeated documentation.

Reply:
That's a really good point. Let me shift some stuff around.

Reply:
Shifted!